cmd/daemon/bootstrap.go

```diff
@@ -0,0 +1,67 @@
+//go:build windows
+
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/daemon/controller"
+	"github.com/openshift/windows-machine-config-operator/pkg/daemon/winsvc"
+	"github.com/openshift/windows-machine-config-operator/pkg/servicescm"
+)
+
+var (
+	bootstrapCmd = &cobra.Command{
+		Use:   "bootstrap",
+		Short: "Starts required Windows Services to bootstrap a Node",
+		Long: "Starts and configures all required Windows Services on an instance according to information given by " +
+			"Windows Service ConfigMaps, resulting in the creation of a new Node object",
+		Run: runBootstrapCmd,
+	}
+	// encodedCMData holds encoded data from the Windows services ConfigMap
+	// TODO: Remove this when the WICD has permissions to get ConfigMaps
+	encodedCMData string
+)
+
+func init() {
+	rootCmd.AddCommand(bootstrapCmd)
+	bootstrapCmd.PersistentFlags().StringVar(&encodedCMData, "services-data", "", "Windows services ConfigMap data")
+	bootstrapCmd.MarkPersistentFlagRequired("services-data")
+}
+
+func runBootstrapCmd(cmd *cobra.Command, args []string) {
+	cmData, err := servicescm.DecodeAndUnmarshall(encodedCMData)
+	if err != nil {
+		klog.Exitf("unable to decode ConfigMap data", err.Error())
+	}
+
+	svcMgr, err := winsvc.NewMgr()
+	if err != nil {
+		klog.Exitf(err.Error())
+	}
+	// TODO: for now, no client is needed for bootstrapping. This will not hold when WICD can get the service ConfigMap
+	serviceController := controller.NewServiceController(ctrl.SetupSignalHandler(), nil, svcMgr, "")
+
+	klog.Info("starting services to bootstrap node")
```

Comment:
Moving this code into a function such as `RunBootstrap` in the controller package would be preferred as this is logic around parsing the services, and not about running the actual exe/flags.